### DISCUSSION TITLE
Cashier (Paddle) button doesn't work when used with this package

### DISCUSSION THREAD

**Discussion by sawirricardo** [spatie/laravel-livewire-wizard#40] [Q&A] [answered]
I have this code
```
class CheckoutStepComponent extends StepComponent
{
public $payLink;
public function mount()
{
$this->payLink =  Auth::user()->newSubscription('default', $premium = 34567)
    ->returnTo(route('home'))
    ->create();
}
}

```

```
// in chekout-step-component.blade.php

<x-paddle-button :url="$payLink" class="px-8 py-4">
    Subscribe
</x-paddle-button>
```

It will do nothing, except we can see the browser url changed to `http://localhost/subscribe?#!`

but putting the paddle button in the regular livewire component works. It shows the modal.

What could be happening?

**Comment by migsAV** [comment, marked as answer]
Try using the Paddle inline option.

https://developer.paddle.com/guides/19115f7f23f75-inline-checkout

In your `chekout-step-component.blade.php` file use the following:
```
	<div class="checkout-container"></div>
	<script type="text/javascript">
		Paddle.Setup({
			vendor: 1234567          // Replace with your Vendor ID.
		});

		Paddle.Checkout.open({
			method: 'inline',
			product: 12345,       // Replace with your Product or Plan ID
			allowQuantity: false,
			disableLogout: true,
			frameTarget: 'checkout-container', // The className of your checkout <div>
			frameInitialHeight: 416,
			frameStyle: 'width:100%; min-width:312px; background-color: transparent; border: none;'    // Please ensure the minimum width is kept at or above 286px with checkout padding disabled, or 312px with checkout padding enabled. See "General" section under "Branded Inline Checkout" below for more information on checkout padding.
		});
	</script>
```

Simpler version

```
@php
    $options = [
        'product' => 1234567,
    ];
@endphp

<x-paddle-checkout :options="$options" class="w-full" />
```

**Comment by adxmcollins** [comment]
This is an old discussion, but I ran into this same problem using Cashier Paddle and Breeze Livewire. I thought I'd leave a comment in case anyone else stumbles across this thread whilst trying to find a fix, just like I did.

The problem in **my case** is that if you're using `wire:navigate` in your application (which Breeze does), code in the `<head>` is not re-evaluated on route change. I suspect that something similar is happening above, although I understand `wire:navigate` wasn't an option back when this thread was started.

Any code which must trigger at route change (analytics, checkout button/element initialisers, etc) should be placed in the `<body>` instead, which is re-evaluated on route change (or you should use one of Livewire's other methods to work around this: https://livewire.laravel.com/docs/navigate#scripts-in-head-are-loaded-once)

I wrote a quick blog post about the fix specifically for Cashier and Breeze also: https://adamcollins.me/journal/fix-laravel-cashier-paddle-buttons-with-breeze-livewire
